### PR TITLE
[feature/RPA-1152] Return output property for completed processes

### PR DESCRIPTION
**Pull request by marcin-depcik**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

---

I agree to the terms of the RunBotics Contributor License Agreement.

### CODE REVIEW

File: runbotics-orchestrator/src/main/java/com/runbotics/web/rest/ProcessResource.java [hunk 1]
```diff
@@ -262,6 +262,22 @@ public ResponseEntity<ProcessDTO> processSetBotSystem(
         );
     }
 
+    @PreAuthorize("@securityService.checkFeatureKeyAccess('" + FeatureKeyConstants.PROCESS_OUTPUT_TYPE_EDIT + "')")
+    @PatchMapping(value = "/processes/{id}/output-type")
+    public ResponseEntity<ProcessDTO> processSetOutputType(
+        @PathVariable(value = "id", required = false) final Long id,
+        @NotNull @RequestBody ProcessDTO processDTO
```
Comment (preludium) [inline runbotics-orchestrator/src/main/java/com/runbotics/web/rest/ProcessResource.java]: imo you should create new DTO to verify fields you want to be included (see is-triggerable and is-attended endpoints)

File: runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20231215807052_changelog.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            <column name="type" type="varchar(50)">
+                <constraints primaryKey="true" nullable="false"/>
+            </column>
+        </createTable>
+        <insert tableName="process_output">
+            <column name="type">JSON_OBJECT</column>
+        </insert>
+        <insert tableName="process_output">
+            <column name="type">PLAIN_TEXT</column>
+        </insert>
```
Comment (preludium) [inline runbotics-orchestrator/src/main/resources/config/liquibase/changelog/20231215807052_changelog.xml]: `JSON` and `TEXT` is sufficient to understand what's inside

File: runbotics/runbotics-common/src/RunboticModdleDescriptor.ts [hunk 3]
```diff
@@ -33,6 +33,12 @@ export const RunboticModdleDescriptor = {
                     type: "Boolean",
                     default: false,
                 },
+                {
+                    name: "processOutput",
+                    isAttr: true,
+                    type: "Boolean",
+                    default: false,
+                },
```
Comment (preludium) [inline runbotics/runbotics-common/src/RunboticModdleDescriptor.ts:41]: what is it for?

File: runbotics/runbotics-desktop/src/core/bpm/CamundaExtension.js [hunk 4]
```diff
@@ -50,8 +50,10 @@ export function Camunda(activity, processEnvironment) {
         //     // behaviour.Service = ServiceImplementation
         // }
         if (activity.type === 'bpmn:ServiceTask') {
+            const disabled = activity.behaviour.disabled;
             activity.environment.runbotic = {
-                disabled: activity.behaviour.disabled,
+                disabled,
+                ...(!disabled && { processOutput: activity.behaviour.processOutput })
```
Comment (preludium) [inline runbotics/runbotics-desktop/src/core/bpm/CamundaExtension.js:56]: is it necessary?

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/BpmnModeler/BpmnModeler.types.ts [hunk 5]
```diff
@@ -291,5 +291,6 @@ export enum ModelerEvent {
     PROPERTIESPANEL_PROVIDERSCHANGED = 'propertiesPanel.providersChanged',
     CANVAS_RESIZED = 'canvas.resized',
     IMPORT_PARSE_COMPLETE = 'import.parse.complete',
+    IMPORT_DONE = 'import.done',
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/BpmnModeler/BpmnModeler.types.ts:294]: we fill attended and trigger details on process import - when does it happen? maybe its better to have output at the same place

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/extensions/customRenderer/CustomRenderer.js [hunk 6]
```diff
@@ -144,7 +147,7 @@ export default class CustomRenderer extends BaseRenderer {
         const label = this.getLabel(element);
         const businessObject = getBusinessObject(element);
         const { disabled } = businessObject;
-
+        // debugger;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/extensions/customRenderer/CustomRenderer.js]: we dont do that here

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/helpers/elementParameters.ts [hunk 7]
```diff
@@ -104,6 +105,14 @@ export type BPMNElement = {
     attachers?: BPMNElement[];
 };
 
+export interface BPMNElementRegistry {
+    [key: BPMNElement['id']]: {
+        element: BPMNElement;
+        gfx: unknown | undefined;
+        secondaryGfx: unknown | undefined;
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/helpers/elementParameters.ts]: russian roulette kind of xd

**Review by preludium — CHANGES_REQUESTED** [review]
shouldn't here be also logic sending output whenever triggered via API

File: runbotics/runbotics-common/src/RunboticModdleDescriptor.ts (hunk 3 above)
Comment (marcin-depcik) [reply runbotics/runbotics-common/src/RunboticModdleDescriptor.ts:41]: it was redundant

File: runbotics/runbotics-desktop/src/core/bpm/CamundaExtension.js (hunk 4 above)
Comment (marcin-depcik) [reply runbotics/runbotics-desktop/src/core/bpm/CamundaExtension.js:56]: I think this is the easiest way to pass form property from each activity of process

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/BpmnModeler/BpmnModeler.types.ts (hunk 5 above)
Comment (marcin-depcik) [reply runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/BpmnModeler/BpmnModeler.types.ts:294]: I think, this event of BPMN Modeler can be misleading, it appears every time we import or save process, but also every time the modeler viewport is loaded

File: runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/helpers/elementForm.ts [hunk 8]
```diff
@@ -78,21 +85,36 @@ export const getFormSchema = (
...
+                'Process.Details.Modeler.ActionPanel.Form.Disabled.Title'
+            )
+        },
+        runFromHere: {
+            type: 'boolean',
+            title: translate(
+                'Process.Details.Modeler.ActionPanel.Form.RunFromHere.Title'
+            )
+        },
+    } as JSONSchema7['properties'];
```
Comment (preludium) [inline runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/helpers/elementForm.ts]: use `satisfies` or `: JSONSchema7['properties']` at the beginning. Avoid using `as` if possible

**Review by preludium — APPROVED** [review]

File: runbotics/runbotics-common/src/RunboticModdleDescriptor.ts (hunk 3 above)
Comment (marcin-depcik) [reply runbotics/runbotics-common/src/RunboticModdleDescriptor.ts:41]: I was wrong, this is necessary to be able persist changes while updating process definition

**Pull request merged**